### PR TITLE
Add clang-tidy config with setting from CLion

### PR DESCRIPTION
**Pull request by jl-wynen**
A config file for clang-tidy. The checks are taken from the default setup of CLion. They seem to generally agree well with the current state of the code but emit warnings about missing `[[nodiscard]]` and `explicit` which was already noted to be missing in Scipp: #1322.

Please suggest changes, I would be happy to make it a bit stricter. But that might mean _a lot_ of warnings.

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
...t-dcl51-cpp,-cert-dcl54-cpp,-cert-dcl59-cpp,-cert-err09-cpp,-cert-err61-cpp,-cert-fio38-c,-cert-msc30-c,-cert-msc32-c,-cert-oop11-cpp,-cert-oop54-cpp,-cert-pos44-c,-cppcoreguidelines-avoid-c-arrays,-cppcoreguidelines-avoid-magic-numbers,-cppcoreguidelines-c-copy-assignment-signature,-cppcoreguidelines-explicit-virtual-functions,-cppcoreguidelines-non-private-member-variables-in-classes,-fuchsia-header-anon-namespaces,-google-readability-braces-around-statements,-google-readability-function-size,-google-readability-namespace-comments,-hicpp-avoid-c-arrays,-hicpp-braces-around-statements,-hicpp-deprecated-headers,-hicpp-explicit-conversions,-hicpp-function-size,-hicpp-invalid-access-moved,-hicpp-member-init,-hicpp-move-const-arg,-hicpp-named-parameter,-hicpp-new-delete-operators,-hicpp-no-array-decay,-hicpp-no-malloc,-hicpp-noexcept-move,-hicpp-special-member-functions,-hicpp-static-assert,-hicpp-undelegated-constructor,-hicpp-uppercase-literal-suffix,-hicpp-use-auto,-hicpp-use-emplace,-hicpp-use-equals-default,-hicpp-use-equals-delete,-hicpp-use-noexcept,-hicpp-use-nullptr,-hicpp-use-override,-hicpp-vararg,google-default-arguments,google-explicit-constructor,google-runtime-operator'
```
Comment (SimonHeybrock) [inline .clang-tidy:2]: Is it possible to write this one per line, for readability?

Comment (jl-wynen) [reply .clang-tidy]: Looks like you can

**Comment by jl-wynen** [comment]
I'd be in favour of activating
```
cppcoreguidelines-avoid-goto
cppcoreguidelines-avoid-non-const-global-variables
cppcoreguidelines-no-malloc
cppcoreguidelines-avoid-c-arrays
cppcoreguidelines-explicit-virtual-functions
google-readability-namespace-comments
```

**Comment by SimonHeybrock** [comment]
> I'd be in favour of activating
> 
> ```
> cppcoreguidelines-avoid-goto
> cppcoreguidelines-avoid-non-const-global-variables
> cppcoreguidelines-no-malloc
> cppcoreguidelines-avoid-c-arrays
> cppcoreguidelines-explicit-virtual-functions
> google-readability-namespace-comments
> ```

👍 I think.

**Pull request merged**